targets/TARGET_STM/gpio_irq_api.c

```diff
@@ -323,7 +323,7 @@ int gpio_irq_init(gpio_irq_t *obj, PinName pin, gpio_irq_handler handler, uint32
     core_util_critical_section_enter();
 
     /* Enable SYSCFG Clock */
-#if !defined(TARGET_STM32WB)
+#if (!defined(TARGET_STM32WB) && !defined(TARGET_STM32WL))
     __HAL_RCC_SYSCFG_CLK_ENABLE();
```

Comment:
In this file, there is 
if defined(DUAL_CORE)
without && (H7) ==> to be added

Good catch @ludoch-stm

Reply:
Done.